skyportal/models.py

```diff
@@ -1203,4 +1207,50 @@ def set_time(self):
     foreign_keys="ClassicalAssignment.requester_id",
 )
 
+
+class Invitation(Base):
+    token = sa.Column(sa.String(), nullable=False, unique=True)
+    groups = relationship(
+        "Group",
+        secondary="group_invitations",
+        cascade="save-update, merge, refresh-expire, expunge",
+        passive_deletes=True,
+    )
+    admin_for_groups = sa.Column(psql.ARRAY(sa.Boolean), nullable=False)
+    user_email = sa.Column(EmailType(), nullable=True)
+    invited_by = relationship(
+        "User",
+        secondary="user_invitations",
+        cascade="save-update, merge, refresh-expire, expunge",
+        passive_deletes=True,
+        uselist=False,
+    )
+    used = sa.Column(sa.Boolean, nullable=False, default=False)
+
+
+GroupInvitation = join_model('group_invitations', Group, Invitation)
+UserInvitation = join_model("user_invitations", User, Invitation)
+
+
+@event.listens_for(Invitation, 'after_insert')
```

Comment:
In the current implementation, this will trigger an internal server error on certain calls to `Group.get` if there is no sengrid API key specified (which there usually won't be, on most SP instances).

Reply:
See comment above -- the changes are to `GroupUser.post`, not `Group.get`

Reply:
@dannygoldstein and a new `Invitation` will only be created if invitations are enabled in the config.